File: packages/agent/src/ast/languages/fixtures/sample.java
package com.demo;

import java.util.List;
import java.util.ArrayList;

/** Greeter type. */
public class Greeter implements Named {
	private int count;
	public static final int MAX = 10;

	public Greeter() {}

	public Greeter(String name) {
		this.count = name.length();
	}

	public void hello() {
		count += 1;
	}

	@Override
	public void name() {}
}

interface Named {
	void name();
}

enum Color {
	RED,
	GREEN;

	void paint() {}
}

record Point(int x, int y) {}

@interface Internal {
	String value() default "";
}

File: packages/agent/src/ast/languages/fixtures/sample.kt
package com.demo

import kotlin.collections.List

/** Greeter */
class Greeter(val name: String) {
	fun hello() {}

	companion object {
		fun create(): Greeter = Greeter("x")
	}
}

internal fun top() {}

object Singleton {
	const val ID = 1
}

enum class Color {
	RED,
	GREEN,
}

interface Named {
	fun name(): String
}

data class Point(val x: Int, val y: Int)

typealias Alias = Greeter
